sql/src/main/java/io/crate/planner/TableStatsService.java

```diff
@@ -50,6 +50,10 @@
 import javax.annotation.Nonnull;
 import java.util.function.Consumer;
 
+/**
+ * Periodically refresh {@link TableStats} based on {@link #refreshInterval}.
+ * The statistics are collected from <code>sys.shards</code> by filtering only on the primary shards.
```

Comment:
Unless this implementation detail is relevant for API consumers I think we shouldn't expose this. Imho it should be possible to change the implementation detail without having to update the class description.

Reply:
hm kind of agree..... since the class could in general calculate and update many kind of stats.